src/controllers/routers/cases.js

```diff
@@ -0,0 +1,140 @@
+/*!
+ * OneSG API Server by TL Bootcamp#6 OneSG Team
+ * Copyright(c) 2020 TechLadies
+ * MIT Licensed
+ */
+
+'use strict';
+
+const {
+  DataError,
+  ForeignKeyViolationError,
+  ValidationError,
+} = require('objection');
+
+const { Case } = require('../../models');
+
+const {
+  errors: { BadRequest, InvalidInput },
+} = require('../../utils');
+
+/**
+ * Sanitize data from client. Call before an insert or an update.
+ */
+
+function sanitize(json) {
+  const cases = json;
+  if (json.caseStatus) {
+    cases.caseStatus = json.caseStatus.toUpperCase().trim();
+  }
+  if (json.pointOfContact) {
+    cases.pointOfContact = json.pointOfContact.trim();
+  }
+  if (json.referenceStatus) {
+    cases.referenceStatus = json.referenceStatus.toUpperCase().trim();
+  }
+  if (json.casePendingReason) {
+    cases.casePendingReason = json.casePendingReason.trim();
+  }
+  if (typeof json.amountRequested === 'string') {
+    cases.amountRequested = parseFloat(json.amountRequested);
+  }
+  if (typeof json.amountGranted === 'string') {
+    cases.amountGranted = parseFloat(json.amountGranted);
+  }
+  if (json.documents) {
+    if (json.documents === null || json.documents === '') {
+      cases.documents = {};
+    }
+  }
+  if (json.refereeId) {
+    cases.refereeId = parseInt(json.refereeId, 10);
+  }
+  if (json.beneficiaryId) {
+    cases.beneficiaryId = parseInt(json.beneficiaryId, 10);
+  }
+  if (json.createdBy) {
+    cases.createdBy = parseInt(json.createdBy, 10);
+  }
+  if (json.updatedBy) {
+    cases.updatedBy = parseInt(json.updatedBy, 10);
+  }
+  return cases;
+}
+
+/**
+ * Retrieve all cases
+ * @param {Request} req
+ * @param {Response} res
+ */
+const getAll = async (req, res) => {
+  const cases = await Case.query().select();
+  return res.status(200).json({ cases });
+};
+
+/**
+ * Create new case
+ * @param {Request} req
+ * @param {Response} res
+ */
+const create = async (req, res, next) => {
+  const newCase = sanitize(req.body);
+  try {
+    const cases = await Case.query().insertGraph(newCase).returning('*');
```

Comment:
1. I thought we discussed that cases and requests need to be a transaction? was there any problem implementing that?
2. for date fields like appliedOn, only the dates are saved in the db (which is as it should be); however the data sent to client  has time included. This should be okay with the create function. But keep a lookout during a normal retrieve.

Reply:
***
They are posted as a transaction using .insertGraph(). An example can be seen in postman where the requests is part of the case req body.

Reply:
updated transaction code to include Case.transaction

Reply:
included a return function in the try block. Async await is required for the outer block as it will return an error without it (UnhandledPromiseRejectionWarning)